File: ui/src/types/Checklist.ts
export interface ChecklistItem {
  title: string
  description: string
}

export interface ChecklistItemResponse extends ChecklistItem {
  id: number
}

export interface Checklist {
  name: string
  description: string
  items?: ChecklistItem[]
}

export interface ChecklistResponse extends Checklist {
  id: number
  createdAt: string
  items?: ChecklistItemResponse[]
}

File: Api/Entities/Checklist.cs
using System.ComponentModel.DataAnnotations;

namespace ChecklistApi.Entities;

public class Checklist
{
    public int Id { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    [MaxLength(100)]
    public string Name { get; set; } = "";

    [MaxLength(1000)]
    public string Description { get; set; } = "";

    public IList<ChecklistItem> Items { get; set; } = new List<ChecklistItem>();
}